src/main/java/de/fraunhofer/aisec/cpg/graph/declarations/FunctionDeclaration.java

```diff
@@ -319,14 +320,26 @@ public boolean equals(Object o) {
     FunctionDeclaration that = (FunctionDeclaration) o;
     return super.equals(that)
         && Objects.equals(body, that.body)
-        && Objects.equals(parameters, that.parameters)
         && Objects.equals(this.getParameters(), that.getParameters())
-        && Objects.equals(throwsTypes, that.throwsTypes)
+        && parameters.size() == that.parameters.size()
+        && IntStream.range(0, parameters.size())
```

Comment:
This seems very error-prone. Can we extract this into a static Function similar to `Objects.equals(this.parameters, that.parameters)`? Something like `PropertyEdge.propertiesEquals(this.parameters, that.parameters)`.

This would allow for easier change of that particular logic, if we ever need to.